app/src/main/java/com/company/rentafield/di/NetworkModule.kt

```diff
@@ -76,66 +78,81 @@ object NetworkModule {
         }.build()
     }
 
-    @Provides
-    @Singleton
-    fun provideService(okHttpClient: OkHttpClient): RetrofitService {
+    private inline fun <reified T> provideRetrofitService(
+        okHttpClient: OkHttpClient,
+        baseUrl: String
+    ): T {
         return Retrofit.Builder()
-            .baseUrl(BASE_URL)
+            .baseUrl(baseUrl)
             .client(okHttpClient)
             .addConverterFactory(Json.asConverterFactory("application/json".toMediaType()))
             .build()
-            .create(RetrofitService::class.java)
+            .create(T::class.java)
     }
 
     @Provides
     @Singleton
     fun provideAiService(okHttpClient: OkHttpClient): RetrofitAiService {
```

Comment:
I think we don't need to pass OkHttpClient here we need to provide it to retrofit direct